### ISSUE TITLE
False positive TooFewArguments for version_compare when one argument is an array key

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#1576]
Seen on dev-master with this snippet:

```php
<?php
function test(string $a, array $x) : bool {
    foreach ($x as $b => $_) {
        // TooFewArguments: Too few arguments for method version_compare - expecting 3 but saw 2
        // This seems to be related to $b being an array key
        if (version_compare($a, $b) < 0) {
            return false;
        }
    }
    return true;
}
test('1', ['2']);
```

Possibly related to #1543

**Comment by SignpostMarv** [comment]
grabbing psalm.dev link for reference: https://psalm.dev/r/b7b4f90587

**Comment by ArSn** [comment]
Can not reproduce this on current master - was it fixed with #2630?

**Comment by TysonAndre** [comment]
The error message for https://psalm.dev/r/b7b4f90587 is now correct. It might have been fixed by that or something else, I think the original issue was also fixed
